### DISCUSSION TITLE
New UMD builds of jbrowse-linear-genome-view embeddable component

### DISCUSSION THREAD

**Discussion by ZWanBing** [GMOD/jbrowse-components#1891] [Q&A] [answered]
hello ! I have used the plugin jbrowse2 with **s3.amazonaws.com/jbrowse.org/jb2_releases/jbrowse-linear-view/jbrowse-linear-view@v0.0.1-beta.0/umd/jbrowse-linear-view.js**
And I noticed that the main release of jbrowse2 can support better indel indications , so could I use this function in the plugin website by a **newer js file**? Thanks a lot♥

**Comment by ZWanBing** [comment]
https://github.com/GMOD/jbrowse-components/issues/1890#issuecomment-816822635
I have used the plugin to make a linear genome viewer on my server so our teammates can view mutation sites with a link directed to a html file with set bam and chromosome position. But the plugin js file is an old version of jbrowse2, so could you provide the recent version of this js script **s3.amazonaws.com/jbrowse.org/jb2_releases/jbrowse-linear-view/jbrowse-linear-view@v0.0.1-beta.0/umd/jbrowse-linear-view.js**? We have urgent needs to get better view on indel.
Thank you so much!

**Comment by garrettjstevens** [comment]
Hi @ZWanBing,

We don't have automated builds yet of the UMD build of our embeddable linear genome view. You can watch this issue for updates on that: https://github.com/GMOD/jbrowse-components/issues/1028. I have gone ahead and done a manual build, though, based off the latest v1.0.0 release. You can access it at:

```
http://s3.amazonaws.com/jbrowse.org/jb2_releases/jbrowse-linear-genome-view/jbrowse-linear-genome-view%40v1.1.0/umd/jbrowse-linear-genome-view.js
```

Note that there has been a change since the version you have been using, and to create the view you will need to use `new JBrowseLinearGenomeView()` (previously it was `new JBrowseLinearView()`).

Also note that this build is a small wrapper around our linear genome view React component [`@jbrowse/react-linear-genome-vew`](https://www.npmjs.com/package/@jbrowse/react-linear-genome-view), which is released regularly, so if it suits your needs you can use that as well.

**Reply by ZWanBing** [reply]
Thanks a lot! I have test it but with the defaultsession it does not work. Below is the website use this plugin:

<html>
  <head>
    <script src="//s3.amazonaws.com/jbrowse.org/jb2_releases/jbrowse-linear-genome-view/jbrowse-linear-genome-view@v1.1.0/umd/jbrowse-linear-genome-view.js"></script>
  </head>

  <body>
    <h1>Test</h1>
    <div id="jbrowse_linear_view"></div>
    <script type="module">
      import assembly from './assembly.js'
      import tracks from './tracks.js'
      const genomeView = new JBrowseLinearGenomeView({
        container: document.getElementById('jbrowse_linear_view'),
        assembly,
        tracks,
        location: 'chr15:66729068',
		defaultSession:{
		name:"defaultSession",
        view:{
		"id": "linearGenomeView",
        "type": "LinearGenomeView",
        "offsetPx": 5608767607,
        "bpPerPx": 0.02,
        "displayedRegions": [{
        "refName": "chr5",
        "start": 0,
        "end": 180915260,
        "reversed": false,
        "assemblyName": "hg19"
		}],
		"tracks": [{
		"id": "ReferenceSequence",
        "type": "ReferenceSequenceTrack",
        "height": 100,
        "configuration": "hg19-ReferenceSequenceTrack"
		},
		{
        "id": "x-j__C_nM",
        "type": "AlignmentsTrack",
         "height": 97,
         "configuration": "2000191.recal-H",
      "PileupTrack": {
        "id": "mrQcOy181Q",
        "type": "PileupTrack",
        "height": 100,
        "configuration": {
          "type": "PileupTrack",
          "trackId": "2000191.recal-H_pileup_xyz",
          "name": "2000191 pileup",
          "assemblyNames": [
            "hg19"
          ],
          "category": [
            "tumor"
          ],
          "adapter": {
            "type": "BamAdapter",
            "bamLocation": {
              "uri": "2000191.recal-H.bam"
            },
            "index": {
              "location": {
                "uri": "2000191.recal-H.bai"
              }
            },
            "sequenceAdapter": {
              "type": "IndexedFastaAdapter",
              "fastaLocation": {
                "uri": "hg19.fasta"
              },
              "faiLocation": {
                "uri": "hg19.fasta.fai"
              }
            }
          },
          "renderers": {
            "PileupRenderer": {
              "type": "PileupRenderer"
            },
            "SvgFeatureRenderer": {
              "type": "SvgFeatureRenderer"
            }
          }
        }
      },
      "SNPCoverageTrack": {
        "id": "DvhupI7m-T",
        "type": "SNPCoverageTrack",
        "height": 40,
        "configuration": {
          "type": "SNPCoverageTrack",
          "trackId": "2000191.recal-H_snpcoverage_xyz",
          "name": "2000191 snpcoverage",
          "assemblyNames": [
            "hg19"
          ],
          "category": [
            "tumor"
          ],
          "adapter": {
            "type": "SNPCoverageAdapter",
            "subadapter": {
              "type": "BamAdapter",
              "bamLocation": {
                "uri": "2000191.recal-H.bam"
              },
              "index": {
                "location": {
                  "uri": "2000191.recal-H.bai"
                }
              },
              "sequenceAdapter": {
                "type": "IndexedFastaAdapter",
                "fastaLocation": {
                  "uri": "hg19.fasta"
                },
                "faiLocation": {
                  "uri": "hg19.fasta.fai"
                }
              }
            }
          },
          "renderers": {
            "SNPCoverageRenderer": {
              "type": "SNPCoverageRenderer"
            }
          }
        },
        "selectedRendering": ""
      },
      "showCoverage": true,
      "showPileup": true
    },
    {
      "id": "EcMTHWIop",
      "type": "AlignmentsTrack",
      "height": 240,
      "configuration": "2000192.recal-H",
      "PileupTrack": {
        "id": "wIZAEuVjB4",
        "type": "PileupTrack",
        "height": 100,
        "configuration": {
          "type": "PileupTrack",
          "trackId": "2000192.recal-H_pileup_xyz",
          "name": "2000192 pileup",
          "assemblyNames": [
            "hg19"
          ],
          "category": [
            "normal"
          ],
          "adapter": {
            "type": "BamAdapter",
            "bamLocation": {
              "uri": "2000192.recal-H.bam"
            },
            "index": {
              "location": {
                "uri": "2000192.recal-H.bai"
              }
            },
            "sequenceAdapter": {
              "type": "IndexedFastaAdapter",
              "fastaLocation": {
                "uri": "hg19.fasta"
              },
              "faiLocation": {
                "uri": "hg19.fasta.fai"
              }
            }
          },
          "renderers": {
            "PileupRenderer": {
              "type": "PileupRenderer"
            },
            "SvgFeatureRenderer": {
              "type": "SvgFeatureRenderer"
            }
          }
        }
      },
      "SNPCoverageTrack": {
        "id": "u8_9Ka-4C7",
        "type": "SNPCoverageTrack",
        "height": 40,
        "configuration": {
          "type": "SNPCoverageTrack",
          "trackId": "2000192.recal-H_snpcoverage_xyz",
          "name": "2000192 snpcoverage",
          "assemblyNames": [
            "hg19"
          ],
          "category": [
            "normal"
          ],
          "adapter": {
            "type": "SNPCoverageAdapter",
            "subadapter": {
              "type": "BamAdapter",
              "bamLocation": {
                "uri": "2000192.recal-H.bam"
              },
              "index": {
                "location": {
                  "uri": "2000192.recal-H.bai"
                }
              },
              "sequenceAdapter": {
                "type": "IndexedFastaAdapter",
                "fastaLocation": {
                  "uri": "hg19.fasta"
                },
                "faiLocation": {
                  "uri": "hg19.fasta.fai"
                }
              }
            }
          },
          "renderers": {
            "SNPCoverageRenderer": {
              "type": "SNPCoverageRenderer"
            }
          }
        },
        "selectedRendering": ""
      },
      "showCoverage": true,
      "showPileup": true
    }
  ],
  "hideHeader": false,
  "hideHeaderOverview": false,
  "trackSelectorType": "hierarchical",
  "showTrackLabels": true,
  "showCenterLine": true
}}
      })
    </script>
  </body>
</html>

What could I do to fix this?

**Reply by garrettjstevens** [reply]
Hi @ZWanBing,

Sorry I didn't see your problem before. It looks like you edited your original comment and replaced it with that message. Just so you know, I don't get email notifications when a comment is edited, so it's probably best to make a new comment instead to make sure I see them.

What problem are you getting when you try to load that defaultSession? I'd probably try re-creating the defaultSession, perhaps using a method like is described here: https://jbrowse.org/jb2/docs/bcc2020_embedding_jbrowse_08_default_session

**Comment by garrettjstevens** [comment]
There's a new manual beta build based on the latest release at:
```
https://s3.amazonaws.com/jbrowse.org/jb2_releases/jbrowse-linear-genome-view/jbrowse-linear-genome-view%40v1.3.2/umd/jbrowse-linear-genome-view.js
```

**Comment by garrettjstevens** [comment, marked as answer]
UMD builds are now officially released starting with v1.3.5. The interface has changed slightly since the beta version from our previous tutorial, and you can see the new tutorial [here](https://jbrowse.org/jb2/docs/tutorials/embed_linear_genome_view/01_introduction).

You can also browse the available build files on [UNPKG](https://unpkg.com/browse/@jbrowse/react-linear-genome-view/dist/).
